test/Ocelot.AcceptanceTests/RoutingTests.cs

```diff
@@ -813,14 +813,14 @@ public void should_return_response_200_with_placeholder_for_final_url_path()
 
         [Theory]
         [Trait("Issue", "748")]
-        [InlineData("/downstream/test/{everything}", "/upstream/test/{everything}", "/upstream/test/1", "/downstream/test/1")]
-        [InlineData("/downstream/test/{everything}", "/upstream/test/{everything}", "/upstream/test/", "/downstream/test/")]
-        [InlineData("/downstream/test/{everything}", "/upstream/test/{everything}", "/upstream/test", "/downstream/test")]
-        [InlineData("/downstream/test/{everything}", "/upstream/test/{everything}", "/upstream/test123", null)]
-        [InlineData("/downstream/{version}/test/{everything}", "/upstream/{version}/test/{everything}", "/upstream/v1/test/123", "/downstream/v1/test/123")]
-        [InlineData("/downstream/{version}/test", "/upstream/{version}/test", "/upstream/v1/test", "/downstream/v1/test")]
-        [InlineData("/downstream/{version}/test", "/upstream/{version}/test", "/upstream/test", null)]
-        public void should_return_correct_downstream_when_omitting_ending_placeholder(string downstreamPathTemplate, string upstreamPathTemplate, string requestURL, string downstreamURL)
+        [InlineData("/downstream/test/{everything}", "/upstream/test/{everything}", "/upstream/test/1", "/downstream/test/1", "?p1=v1&p2=v2&something-else")]
+        [InlineData("/downstream/test/{everything}", "/upstream/test/{everything}", "/upstream/test/", "/downstream/test/", "?p1=v1&p2=v2&something-else")]
+        [InlineData("/downstream/test/{everything}", "/upstream/test/{everything}", "/upstream/test", "/downstream/test", "?p1=v1&p2=v2&something-else")]
+        [InlineData("/downstream/test/{everything}", "/upstream/test/{everything}", "/upstream/test123", null, null)]
+        [InlineData("/downstream/{version}/test/{everything}", "/upstream/{version}/test/{everything}", "/upstream/v1/test/123", "/downstream/v1/test/123", "?p1=v1&p2=v2&something-else")]
+        [InlineData("/downstream/{version}/test", "/upstream/{version}/test", "/upstream/v1/test", "/downstream/v1/test", "?p1=v1&p2=v2&something-else")]
+        [InlineData("/downstream/{version}/test", "/upstream/{version}/test", "/upstream/test", null, null)]
+        public void should_return_correct_downstream_when_omitting_ending_placeholder(string downstreamPathTemplate, string upstreamPathTemplate, string requestURL, string downstreamURL, string queryString)

```

Comment:
Exactly it was my idea to fix... but it was too late for me yesterday's tonight 😸 
Thanks for fixing!

Reply:
Seems our acceptance tests confirm that nothing broken in Catch-All feature, and we've introduced new empty placeholders for Catch-All too. Query String Catch-All should work too.
Great!

`^(?i)/contracts/\?.*$`